tests/h/search/index_test.py

```diff
@@ -99,7 +99,7 @@ def test_index_indexes_all_annotations_to_es(self, db_session, indexer, matchers
         indexer.index()
 
         streaming_bulk.assert_called_once_with(
-            indexer.es_client.conn, matchers.iterable_with([ann_1, ann_2]),
+            indexer.es_client.conn, matchers.iterable_with(matchers.unordered_list([ann_1, ann_2])),
```

Comment:
The fix here relies on the fact that `iterable_with()` does `list(other) == self.items`. That is, it turns the generator that `streaming_bulk()` was called with (`other`) into a list (my theory is this list will sometimes be `[ann_1, ann_2]` and sometimes come out as `[ann_2, ann_1]`) and then compares that list to `self.items` which is `matchers.unordered_list([ann_1, ann_2])`. This comparison will be `True` regardless of the order of the lists.

https://github.com/hypothesis/h/blob/66f83ba23bc657a440e4a341c9593023be85cba5/tests/common/matchers.py#L65-L75